src/grpc/iot_config/hpr_route_stream_worker.erl

```diff
@@ -205,40 +209,96 @@ terminate(_Reason, _State) ->
 %% ------------------------------------------------------------------
 
 -spec process_route_stream_res(
-    RouteStreamRes :: hpr_route_stream_res:res(), Path :: path()
+    RouteStreamRes :: hpr_route_stream_res:action(),
+    Data ::
+        {route, hpr_route:route()}
+        | {eui_pair, hpr_eui_pair:eui_pair()}
+        | {devaddr_range, hpr_devaddr_range:devaddr_range()}
 ) -> ok.
-process_route_stream_res(RouteStreamRes, State) ->
-    Route = hpr_route_stream_res:route(RouteStreamRes),
-    case hpr_route_stream_res:action(RouteStreamRes) of
-        delete ->
-            hpr_route_ets:delete(Route);
-        _ ->
-            hpr_route_ets:insert(Route)
-    end,
-    case maybe_cache_response(RouteStreamRes, State) of
-        {error, Reason} -> lager:error("failed to write to file ~p", [Reason]);
-        ok -> ok
-    end.
+process_route_stream_res(add, {route, Route}) ->
+    hpr_route_ets:insert_route(Route);
+process_route_stream_res(add, {eui_pair, EUIPair}) ->
+    hpr_route_ets:insert_eui_pair(EUIPair);
+process_route_stream_res(add, {devaddr_range, DevAddrRange}) ->
+    hpr_route_ets:insert_devaddr_range(DevAddrRange);
+process_route_stream_res(remove, {route, Route}) ->
+    hpr_route_ets:delete_route(Route);
+process_route_stream_res(remove, {eui_pair, EUIPair}) ->
+    hpr_route_ets:delete_eui_pair(EUIPair);
+process_route_stream_res(remove, {devaddr_range, DevAddrRange}) ->
+    hpr_route_ets:delete_devaddr_range(DevAddrRange).
 
--spec maybe_cache_response(RouteStreamRes :: hpr_route_stream_res:res(), path()) ->
-    ok | {error, any()}.
-maybe_cache_response(_RouteStreamRes, undefined) ->
+-spec maybe_cache_response(
+    Path :: path(),
+    RouteStreamRes :: hpr_route_stream_res:action(),
+    Data ::
+        {route, hpr_route:route()}
+        | {eui_pair, hpr_eui_pair:eui_pair()}
+        | {devaddr_range, hpr_devaddr_range:devaddr_range()}
+) -> ok.
+maybe_cache_response(undefined, _Action, _Data) ->
     ok;
-maybe_cache_response(RouteStreamRes, Path) ->
+maybe_cache_response(Path, add, {route, Route}) ->
```

Comment:
Can we debounce this somehow?
We're going to slam the disk everytime we get a new connection or bring the box up.